client/ping.go

```diff
@@ -61,6 +63,13 @@ func parsePingResponse(cli *Client, resp serverResponse) (types.Ping, error) {
 	if bv := resp.header.Get("Builder-Version"); bv != "" {
 		ping.BuilderVersion = types.BuilderVersion(bv)
 	}
+	if si := resp.header.Get("Swarm"); si != "" {
+		parts := strings.SplitN(si, " ", 2)
+		ping.SwarmStatus = &swarm.Status{
+			NodeState:        swarm.LocalNodeState(parts[0]),
+			ControlAvailable: len(parts) == 2 && parts[1] == "(manager)",
```

Comment:
This seems a little bit fiddly -- we're essentially parsing what's effectively a human-readable string back into separate parts? :grimacing:

Reply:
Yes, agreed. I've been going back-and-forth on what the best approach would be; we could make the format itself more "parseable" (`<status>/<role>`) or two separate headers (I wasn't sure about the separate headers, because we won't be able to set the `role` until the status is also `available` (IIRC).